src/libos/src/fs/syscalls.rs

```diff
@@ -700,3 +700,45 @@ pub fn do_flock(fd: FileDesc, operation: i32) -> Result<isize> {
     file_ops::do_flock(fd, flock_ops)?;
     Ok(0)
 }
+
+pub fn do_utime(path: *const i8, times_u: *const utimbuf_t) -> Result<isize> {
+    let utimes_tuple = if !times_u.is_null() {
+        Some(utimes_timespec_tuple_t::from_utimbuf_ptr(times_u)?)
+    } else {
+        None
```

Comment:
"If times is NULL, then the access and modification times of the file are set to the current time." 
Should we set the utimes_tuple with current time instead of using Option?

Reply:
If the `times_u` is null, the `get_utimes(...)` in `file_ops/utimes.rs` will be called with `times : Option<...> = None`, then it will return:
```
Ok(TimespecTuple {
    atime: Some(now),
    mtime: Some(now),
})
```

Reply:
What I mean is: if we could do the check here, then we do not need to use utimes_timespec_tuple_t instead of Option<utimes_timespec_tuple_t> as the parameter of do_utimes.
BTW: Can we use get_utimes here? If we do this, we may be able to return error earlier.

Reply:
I have refactored the code, the diff can be seen in https://github.com/occlum/occlum/pull/846/files. If we think it is better, I will apply such changes to current PR.

Reply:
Done.